sys-firmware/edk2-ovmf/edk2-ovmf-202405.ebuild

```
@@ -0,0 +1,169 @@
+virtual machines. The firmware is located under
+	/usr/share/edk2-ovmf/OVMF_CODE.fd
+	/usr/share/edk2-ovmf/OVMF_VARS.fd
+	/usr/share/edk2-ovmf/OVMF_CODE.secboot.fd
+
+If USE=binary is enabled, we also install an OVMF variables file (coming from
+fedora) that contains secureboot default keys
+
+	/usr/share/edk2-ovmf/OVMF_VARS.secboot.fd
+
+If you have compiled this package by hand, you need to either populate all
+necessary EFI variables by hand by booting
+	/usr/share/edk2-ovmf/UefiShell.(iso|img)
+or creating OVMF_VARS.secboot.fd by hand:
+	https://github.com/puiterwijk/qemu-ovmf-secureboot
+
+The firmware does not support csm (due to no free csm implementation
+available). If you need a firmware with csm support you have to download
+one for yourself. Firmware blobs are commonly labeled
+	OVMF{,_CODE,_VARS}-with-csm.fd
+
+In order to use the firmware you can run qemu the following way
+
+	$ qemu-system-x86_64 \
+		-drive file=/usr/share/edk2-ovmf/OVMF.fd,if=pflash,format=raw,unit=0,readonly=on \
+		..."
+
+pkg_setup() {
+	python-any-r1_pkg_setup
+	secureboot_pkg_setup
+}
+
+src_prepare() {
+	# Bundled submodules
+	cp -rl "${WORKDIR}/openssl-${BUNDLED_OPENSSL_SUBMODULE_SHA}"/* "CryptoPkg/Library/OpensslLib/openssl/" \
+		|| die "copying openssl failed"
+	cp -rl "${WORKDIR}/brotli-${BUNDLED_BROTLI_SUBMODULE_SHA}"/* "BaseTools/Source/C/BrotliCompress/brotli/" \
+		|| die "copying brotli failed"
+	cp -rl "${WORKDIR}/brotli-${BUNDLED_BROTLI_SUBMODULE_SHA}"/* \
+		"MdeModulePkg/Library/BrotliCustomDecompressLib/brotli/" || die "copying brotli failed"
+	cp -rl "${WORKDIR}/public-mipi-sys-t-${BUNDLED_MIPI_SYS_T_SUBMODULE_SHA}"/* "MdePkg/Library/MipiSysTLib/mipisyst/" \
+		|| die "copying mipi-sys-t failed"
+	cp -rl "${WORKDIR}/mbedtls-${BUNDLED_MBEDTLS_SUBMODULE_SHA}"/* "CryptoPkg/Library/MbedTlsLib/mbedtls/" \
+		|| die "copying mbedtls failed"
+	cp -rl "${WORKDIR}/libspdm-${BUNDLED_LIBSPDM_SUBMODULE_SHA}"/* "SecurityPkg/DeviceSecurity/SpdmLib/libspdm" \
+		|| die "copying libspdm failed"
+
+	sed -i -r \
+		-e "/function SetupPython3/,/\}/{s,\\\$\(whereis python3\),${EPYTHON},g}" \
+		"${S}"/edksetup.sh || die "Fixing for correct Python3 support failed"
+
+	default
+}
+
+src_compile() {
+	TARGET_ARCH=X64
+	TARGET_NAME=RELEASE
+	TARGET_TOOLS=GCC5
+
+	BUILD_FLAGS="-D TLS_ENABLE \
+		-D HTTP_BOOT_ENABLE \
+		-D NETWORK_IP6_ENABLE \
+		-D TPM_ENABLE \
+		-D TPM2_ENABLE -D TPM2_CONFIG_ENABLE \
+		-D FD_SIZE_2MB"
+
+	SECUREBOOT_BUILD_FLAGS="${BUILD_FLAGS} \
+		-D SECURE_BOOT_ENABLE \
+		-D SMM_REQUIRE \
+		-D EXCLUDE_SHELL_FROM_FD"
+
+	export LDFLAGS="-z notext"
+	export EXTRA_LDFLAGS="-z notext"
+	export DLINK_FLAGS="-z notext"
+
+	emake ARCH=${TARGET_ARCH} -C BaseTools
+
+	. ./edksetup.sh
+
+	# Build all EFI firmware blobs:
+
+	mkdir -p ovmf
+
+	./OvmfPkg/build.sh \
+		-a "${TARGET_ARCH}" -b "${TARGET_NAME}" -t "${TARGET_TOOLS}" \
+		${BUILD_FLAGS} || die "OvmfPkg/build.sh failed"
+
+	cp Build/OvmfX64/*/FV/OVMF_*.fd ovmf/
+	rm -rf Build/OvmfX64
```

We don't use `-f` for predictable files.
```suggestion
	rm -r Build/OvmfX64 || die
```